src/security-council-mgmt/governors/modules/SecurityCouncilNomineeElectionGovernorIndexingTiming.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "../../interfaces/ISecurityCouncilManager.sol";
+
+import "@openzeppelin/contracts-upgradeable/governance/GovernorUpgradeable.sol";
+import "lib/solady/src/utils/DateTimeLib.sol";
+
+abstract contract SecurityCouncilNomineeElectionGovernorIndexingTiming is Initializable, GovernorUpgradeable {
+    /// @notice Date struct for convenience
+    struct Date {
+        uint256 year;
+        uint256 month;
+        uint256 day;
+        uint256 hour;
+    }
+
+    /// @notice First election start date
+    Date public firstNominationStartDate;
+
+    /// @notice Duration of the nominee vetting period (expressed in blocks)
+    /// @dev    This is the amount of time after voting ends that the nomineeVetter can exclude noncompliant nominees
+    uint256 public nomineeVettingDuration;
+
+    /// @notice Number of elections created
+    uint256 public electionCount;
+
+    function __SecurityCouncilNomineeElectionGovernorIndexingTiming_init(
+        Date memory _firstNominationStartDate,
+        uint256 _nomineeVettingDuration
+    ) internal onlyInitializing {
+        require(
+            DateTimeLib.isSupportedDateTime({
+                year: _firstNominationStartDate.year,
+                month: _firstNominationStartDate.month,
+                day: _firstNominationStartDate.day,
+                hour: _firstNominationStartDate.hour,
+                minute: 0,
+                second: 0
+            }),
+            "SecurityCouncilNomineeElectionGovernor: Invalid first nomination start date"
+        );
+
+        // make sure the start date is in the future
+        uint256 startTimestamp = DateTimeLib.dateTimeToTimestamp({
+            year: _firstNominationStartDate.year,
+            month: _firstNominationStartDate.month,
+            day: _firstNominationStartDate.day,
+            hour: _firstNominationStartDate.hour,
+            minute: 0,
+            second: 0
+        });
+
+        require(
+            startTimestamp > block.timestamp,
+            "SecurityCouncilNomineeElectionGovernor: First nomination start date must be in the future"
+        );
+
+        firstNominationStartDate = _firstNominationStartDate;
+        nomineeVettingDuration = _nomineeVettingDuration;
+    }
+
+    /************** view/pure functions **************/
+
+    /// @notice Returns the deadline for the nominee vetting period for a given `proposalId`
+    function proposalVettingDeadline(uint256 proposalId) public view returns (uint256) {
+        return proposalDeadline(proposalId) + nomineeVettingDuration;
+    }
+
+    /// @notice Returns the start timestamp of an election
+    /// @param firstElection The start date of the first election
+    /// @param electionIndex The index of the election
+    function electionToTimestamp(Date memory firstElection, uint256 electionIndex)
```

Comment:
You could also/instead have a `function electionToTimestamp(uint256 electionIndex) view returns(uin256)` which uses the stored first date